Library/Homebrew/cmd/link.rb

```diff
@@ -27,16 +27,27 @@ def link
 
     ARGV.kegs.each do |keg|
       keg_only = Formulary.keg_only?(keg.rack)
-      if HOMEBREW_PREFIX.to_s == "/usr/local" && keg_only &&
-         keg.name.start_with?("openssl", "libressl")
-        opoo <<~EOS
-          Refusing to link: #{keg.name}
-          Linking keg-only #{keg.name} means you may end up linking against the insecure,
-          deprecated system OpenSSL while using the headers from Homebrew's #{keg.name}.
-          Instead, pass the full include/library paths to your compiler e.g.:
-            -I#{HOMEBREW_PREFIX}/opt/#{keg.name}/include -L#{HOMEBREW_PREFIX}/opt/#{keg.name}/lib
-        EOS
-        next
+      if HOMEBREW_PREFIX.to_s == "/usr/local" && keg_only
+        if keg.name.start_with?("openssl", "libressl")
+          opoo <<~EOS
+            Refusing to link: #{keg.name}
+            Linking keg-only #{keg.name} means you may end up linking against the insecure,
+            deprecated system OpenSSL while using the headers from Homebrew's #{keg.name}.
+            Instead, pass the full include/library paths to your compiler e.g.:
+              -I#{HOMEBREW_PREFIX}/opt/#{keg.name}/include -L#{HOMEBREW_PREFIX}/opt/#{keg.name}/lib
+          EOS
+          next
+        elsif (MacOS.version >= :mojave ||
+               MacOS::Xcode.version >= "10.0" ||
+               MacOS::CLT.version >= "10.0") &&
+              dep_f.keg_only_reason.reason == :provided_by_macos
```

Comment:
I think we also want to check `shadowed_by_macos` too since finding the `libedit` headers may be an issue too.

Reply:
👍 but let's wait until we see that in the wild first?